nvdaHelper/remote/winword.cpp

```diff
@@ -525,9 +526,18 @@ void generateXMLAttribsForFormatting(IDispatch* pDispatchRange, int startOffset,
 	if((formatConfig&formatConfig_reportLineNumber)&&(_com_dispatch_raw_method(pDispatchRange,wdDISPID_RANGE_INFORMATION,DISPATCH_PROPERTYGET,VT_I4,&iVal,L"\x0003",wdFirstCharacterLineNumber)==S_OK)) {
 		formatAttribsStream<<L"line-number=\""<<iVal<<L"\" ";
 	}
-	if((formatConfig&formatConfig_reportAlignment)||(formatConfig&formatConfig_reportParagraphIndentation)||(formatConfig&formatConfig_reportLineSpacing)) {
+	if((formatConfig&formatConfig_reportColor)||(formatConfig&formatConfig_reportAlignment)||(formatConfig&formatConfig_reportParagraphIndentation)||(formatConfig&formatConfig_reportLineSpacing)) {
 		IDispatchPtr pDispatchParagraphFormat=NULL;
 		if(_com_dispatch_raw_propget(pDispatchRange,wdDISPID_RANGE_PARAGRAPHFORMAT,VT_DISPATCH,&pDispatchParagraphFormat)==S_OK&&pDispatchParagraphFormat) {
+			if(formatConfig&formatConfig_reportColor) {
+				IDispatchPtr pDispatchParagraphFormatShading=NULL;
+				if(_com_dispatch_raw_propget(pDispatchParagraphFormat,wdDISPID_PARAGRAPHFORMAT_SHADING,VT_DISPATCH,&pDispatchParagraphFormatShading)==S_OK&&pDispatchParagraphFormatShading) {
+					int bgColor=0;
+					if(_com_dispatch_raw_propget(pDispatchParagraphFormatShading,wdDISPID_SHADING_BACKGROUNDPATTERNCOLOR,VT_I4,&bgColor)==S_OK) {
+						formatAttribsStream<<L"background-color=\"" << bgColor <<L"\" ";
+					}
+				}

```

Comment:
please try and lint all of these changes with appropriate spacing around operators.

Reply:
I did it manually, I don't know how to run lint on .cpp files (runlint.bat seems to ignore them).

Reply:
yes, we don't have an automatic linting check for our C++